### PR TITLE
[MAD][DATA1018][Lab intro to ml] Alberto García Cobo

### PR DESCRIPTION
**Pull request by albertogcmr**
Lab intro to ml

### CODE REVIEW

File: module-3/lab-intro-to-ml/your-code/main.ipynb [hunk 1]
```diff
@@ -2905,7 +2905,7 @@
     "                  'VisibilityLowMiles', 'WindHighMPH', 'WindAvgMPH', 'WindGustMPH', \n",
```
Comment (Gobinde43) [inline module-3/lab-intro-to-ml/your-code/main.ipynb]: - Using apply to change data type in wrong columns
```Python
austin[wrong_type_columns] = austin[wrong_type_columns].apply(lambda col: pd.to_numeric(col, errors='coerce'))
```
- Use size method instead of len in the following and check the differences. 
```Python 
number_rows_austin = len(df_austin)
number_rows_missing = len(missing_values)
````
- You don't need all of this 
```Python
missing_rows_count = {}
for col in df_austin.columns: 
    missing_rows_count[col] = df_austin[col].isna().sum()
missing_rows_count

#Just doing this you get the missing rows per column. 
missing_values.isna().sum()
```
- You don't need all of this 
```Python
missing_rows_count_austin = {}
for col in df_austin.columns: 
    missing_rows_count_austin[col] = df_austin[col].isna().sum()
display(missing_rows_count_austin)
#######################
missing_rows_count_austin_fixed = {}
for col in austin_fixed.columns: 
    missing_rows_count_austin_fixed[col] = austin_fixed[col].isna().sum()
display(missing_rows_count_austin_fixed)

#Only checking the following throw the wanted results 
austin_fixed[austin_fixed.isnull().any(axis=1)]
````
- "Next, populate the actual values in the dummy columns of  austin_fixed". Other way of doing it: 
```Python
for event in event_list:
    austin_fixed[event] = austin_fixed['Events'].str.contains(event).astype(int)
```
- "In the cell below, convert the Date column values from string to numeric values using toordinal()." Other way of doing it: 
```Python
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- "Your y should be a subset of austin_fixed containing one column TempAvgF." You can do something like the following instead of writing down each column label. The training should work 
```Python
response_col_names = list(austin_fixed.columns.values)
response_col_names.remove('TempAvgF')
X = austin_fixed[response_col_names]
y = austin_fixed['TempAvgF']
```
-If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
```
